modules/docs/src/main/mdoc/docs/combinators.md

```diff
@@ -141,6 +147,88 @@ val io = retryingOnAllErrors(
 io.unsafeRunSync()
 ```
 
+## `retryingOnFailuresAndSomeErrors`
+
+This is a combination of `retryingM` and `retryingOnSomeErrors`. It allows you to specify failure
+conditions for both the results of your results as well as errors that can occur.
+
+To use `retryingOnFailuresAndSomeErrors`, you need to pass in predicates that decide
+whether a given error or result is worth retrying. 
+
+The API (modulo some type-inference trickery) looks like this:
+
+```scala
+def retryingOnFailuresAndSomeErrors[M[_]: Monad, A, E](policy: RetryPolicy[M],
+                                                       wasSuccessful: A => Boolean,
+                                                       isWorthRetrying: E => Boolean,
+                                                       onFailure: (A, RetryDetails) => M[Unit],
+                                                       onError: (E, RetryDetails) => M[Unit])
+                                                       (action: => M[A]): M[A]
+```
+
+You need to pass in:
+
+* a retry policy
+* a predicate that decides whether a result is worth retrying
```

Comment:
I would phrase this as `a predicate that decides whether the operation was successful`